addons/purchase/models/product.py

```diff
@@ -79,6 +79,11 @@ def action_view_po(self):
 class ProductSupplierinfo(models.Model):
     _inherit = "product.supplierinfo"
 
+    discount = fields.Float(
+        string="Discount (%)",
+        digits='Discount',
+        readonly=False)
+
```

Comment:
You could add it in product module since the model is defined there. If it's use outside purchase, it could be useful too